### PR TITLE
fix: Fixed naming issue in order items for order creation and updation form.

### PR DESCRIPTION
**Pull request by Kashif-Kamran**
For the order updation the apporach is as following:
- before loading the edit form the order should be retrived from the server and then populated in the form. 

## Summary by CodeRabbit

- **New Features**
	- Introduced a loading state in the Edit Order Form for improved user experience.
	- Added a new Order Form Skeleton component for displaying a loading placeholder.

- **Improvements**
	- Enhanced order item structure in the API response for better data visualization.
	- Updated product name rendering logic to ensure robustness in the Order Form.
	- Improved handling of customer selection in the Order Form.

- **Bug Fixes**
	- Adjusted input handling for quantity and price to ensure correct data types.

- **Schema Updates**
	- Modified the Order Schema to include a new optional variant field and customer object for enhanced visualization.

- **Interface Changes**
	- Renamed property in the IProductVariant interface to reflect a singular product association.

### CODE REVIEW

File: apps/web/src/routes/orders/order-list/components/forms/edit-order-form.tsx [hunk 1]
```diff
@@ -32,6 +34,6 @@ export const EditOrderForm: React.FC<{ order: IOrder; onClose: () => void }> = (
       }
     );
   };
-
-  return <OrderForm order={order} onSubmit={handleSubmit} onClose={onClose} isPending={isPending} />;
+  if (isLoading) return <OrderFormSkeleton />;
```
Comment (zainzafar90) [inline apps/web/src/routes/orders/order-list/components/forms/edit-order-form.tsx:37]: super-nit: but I would love some breathing space between the lines, its not a part of the review but it feels more easier on the eyes

File: apps/web/src/routes/orders/order-list/components/forms/order-form.tsx [hunk 2]
```diff
@@ -190,6 +190,11 @@ const OrderSummary = ({ form }: { form: UseFormReturn<OrderFormData> }) => {
 const CustomerSummary = ({ form }: { form: UseFormReturn<OrderFormData> }) => {
   const [isCustomerDialogOpen, openCustomerDialog, closeCustomerDialog] = useToggleState();
   const [selectedCustomer, setSelectedCustomer] = useState<ICustomer | null>(null);
+  const customer = useWatch({ name: 'customer' }) as ICustomer;
+
+  useEffect(() => {
```
Comment (zainzafar90) [inline apps/web/src/routes/orders/order-list/components/forms/order-form.tsx:195]: I think we can the leave useEffect out and pass in the customer directly right? As it gets loaded on mount

File: apps/web/src/routes/orders/order-list/components/forms/edit-order-form.tsx (hunk 1 above)
Comment (Kashif-Kamran) [reply apps/web/src/routes/orders/order-list/components/forms/edit-order-form.tsx:37]: Okay zain bhai, on it

File: apps/web/src/routes/orders/order-list/components/forms/order-form.tsx [hunk 3]
```diff
@@ -289,7 +294,9 @@ const OrderItems = ({ form }: { form: UseFormReturn<OrderFormData> }) => {
 
           {fields.map((field, index) => (
             <TableRow key={field.id}>
-              <TableCell>{selectedVariants[index]?.name}</TableCell>
+              <TableCell>
+                {selectedVariants[index]?.product?.name || field.variant?.product?.name || 'Unknown Produt'}
```
Comment (zainzafar90) [inline apps/web/src/routes/orders/order-list/components/forms/order-form.tsx:298]: I think both product name and variant are important. 

Plus why is index based product and the field based product name? Can we clean this?

**Review by zainzafar90 — CHANGES_REQUESTED** [review]
Left few comments

**Pull request merged**